### ISSUE TITLE
Collisions sometimes importing with weird shapes

### ISSUE THREAD

**Issue by Josh1billion** [Seanba/SuperTiled2Unity#101]
Unity version: 2019.1.6f1
SuperTiled2Unity version: 1.5.8 (have observed it on much older versions as well before upgrading a couple of months ago)

I'm running into an issue where sometimes the collider that gets generated results in weird shapes, where collisions exist where they're not supposed to.

Here's what my map looks like:

![image](https://user-images.githubusercontent.com/934871/67149380-4c303f00-f270-11e9-8401-0eeeb62cd96c.png)

For simplicity, instead of making individual content tiles (walls, bookshelves, etc.) collidable, I just created a tileset with a red "X" and made that solid.  At runtime, my game looks for that layer and makes it invisible, keeping the collisions.

In other words, the tileset for collisions looks like this:

![image](https://user-images.githubusercontent.com/934871/67149528-cf05c980-f271-11e9-834d-db97d2ccf846.png)

What I'm expecting is that only the areas of the map marked with red X's are solid/untraversable.

But what I'm seeing is that, sometimes, this happens:
![image](https://user-images.githubusercontent.com/934871/67149541-04aab280-f272-11e9-83e6-fc1b71c5634b.png)

As you can see, there's an area on the left side that's solid/collidable that shouldn't be.  Just an invisible wall.

Here's what it looks like with the other layers turned off, to make it easier to see:
![image](https://user-images.githubusercontent.com/934871/67149568-45a2c700-f272-11e9-941b-dc65637c080a.png)

My collision settings look like this:
![image](https://user-images.githubusercontent.com/934871/67149573-5f440e80-f272-11e9-8924-00caa2c00d3c.png)

I've confirmed my Tiled map is set up to Snap To Pixels, in case that matters (saw that mentioned in issue #96).

**Comment by Seanba** [comment]
Hi there, @Josh1billion. Can you send me your project to look at?

**Comment by Josh1billion** [comment]
@Seanba unfortunately it's a pretty huge project (a couple of gigs).  Is there anything specific I can screenshot, video walkthrough, etc. that would be useful to see?

**Comment by Seanba** [comment]
Would it be possible to give me the map and tileset files that have this error? You can right-click on your `Assets` directory unity and select `Export Package...` and from there you can narrow down how many files you want exported.

**Comment by Seanba** [comment]
@Josh1billion if you're still having issues you might find a new feature with version 1.8.0 of SuperTiled2Unity helpful: You can choose a single Tiled map and it with all of its dependencies (tilesets and textures, for instance) will automatically be selected for export into a Unity Package ...

![image](https://user-images.githubusercontent.com/6980630/67629346-33acbf80-f842-11e9-8ee7-a317ecb088e6.png)

![image](https://user-images.githubusercontent.com/6980630/67629348-38717380-f842-11e9-8843-0d8eae878d4f.png)

If you send me the Unity package of the problematic map I will be happy to debug further.

**Comment by Josh1billion** [comment]
Hey, sorry for the late response.  Haven't had time to work on my game much in the past couple of weeks.  I'll try that out soon.

**Comment by Josh1billion** [comment]
Hi, sorry for the long wait on this.  I finally figured out the issue.  It's not Super Tiled2Unity's fault.

It turns out that my tileset had a tile in it that was invisible (i.e., transparent in every pixel), so I didn't realize I had accidentally placed it in my map sometimes.  That normally wouldn't be a problem, except.. I had also accidentally given that tile a solid collision.

Thus, I had randomly placed collidable tiles throughout my map without ever noticing.  Oops.  I only realized it when I tried "erasing" on a part of a map that had an incorrect collision, and observed that the problem-causing collision shape went away.

Sorry for wasting your time with this.  Just a classic case of user error. :)
